### DISCUSSION TITLE
Super Notes feature requests

### DISCUSSION THREAD

**Discussion by jrcacd** [standardnotes/forum#3008] [Suggestions]
- [x] Add option to highlight text (yellow color)
- [ ] Ability to resize images

**Comment by effieeee** [comment]
@chloeso (https://github.com/standardnotes/forum/issues/2306)
- [ ] LaTeX math equations

**Reply by nunesgh** [reply]
@effieeee, I think KaTeX (@Teskann, https://github.com/standardnotes/forum/issues/3782) would be relevant here.

**Comment by ashutoshsaboo** [comment]
Big +1 for LaTeX math equations. Greatly missing those from the milkdown editor. From Lexical's playground (https://playground.lexical.dev/), Looks like lexical has already built out of the box support as well for LaTeX - both by the menu popup (like tables) as well as by entering equations within `$` placeholders for a more faster experience - exactly similar how Milkdown also supports it already. If both can be enabled for SN super notes, that'd be really _super_ awesome! :) @effieeee

**Comment by shompoe** [comment]
Ability to convert supernotes back to markdown or html without having to export to an external text file.

**Comment by shompoe** [comment]
Horizontally scrolling tables which auto-size to content rather than screen size.

**Comment by viharsheth** [comment]
Listed doesn't show an image if a Super Note is published to the blog. Would love for Super Notes to show up as written on Listed.

**Comment by shompoe** [comment]
"Show Markdown" option does not convert tables.

**Comment by cuathro** [comment]
- [ ] Search (with highlighting) within the note

**Comment by PovilasID** [comment]
OICD or LDAP auth for selfhosted option

**Comment by effieeee** [comment]
[Supernotes: Change the "add link" button's behavior](https://github.com/standardnotes/forum/issues/2372)

**Comment by THEOCKID** [comment]
Android Widgets.  Kind of deal breaker, sorry

**Comment by danielcolquitt** [comment]
Word count, reading time, and other stats for Super Notes.

**Comment by erlandh** [comment]
- [ ] Add or remove rows or columns in a table
- [ ] Resize columns in tables

**Comment by jomra** [comment]
This isn't available in any of the alternatives I've tried, but in-note links, such as to a reference in a citations section, would be super useful for decluttering notes. Otherwise, you have to either link inline (which is an issue because of link rot), manually add numbered footnotes, or clutter up the note with inline references.

Edit - When I say references, I don't necessarily mean a formal academic citation. Just something with more info than a bare URL.

**Comment by shompoe** [comment]
Ability to adjust paragraph spacing so as to eliminate the need for extra line breaks.

Presently the spacing between list items is defined by the line spacing setting in preferences, this should also be defined by a separate setting for paragraph spacing.

**Comment by SpaceRaccoonFluff** [comment]
Table of Contents. Click to jump to a section. Notion has it. 

My notes are divided into sections each having large titles to make them easy to spot. The issue however is once your notes become as large as a book itself, scrolling to a certain section of your notes becomes a serious daily pain. Being able to just write the titles of my note sections at the very top of my notes and click on them to be taken to my specific section would be a huge productivity boost since i'm constantly switching between my browser and Standard Notes to add to certain sections of my notes.

**Comment by SpaceRaccoonFluff** [comment]
The ONLY way I take a note is by using bullet points. I make severe use of the indent feature of indented bullet points. The only issue is when you have a VERY large blob of a bullet point with multiple indents, Standard Notes bullet points are pretty ugly compared to the competition. It is so HARD to tell indentations apart! Most of the other editors are bad as well but Super editor takes it to a different level of bad UX. 

I was using NotesNook and they do bullet points BEAUTIFULLY. They have something called "Outline list" which was what I was using. It makes looking back at my notes so much more elegant of an experience. These are the types of bullet points I would love to see in Standard Notes as well.
![Screenshot from 2023-06-30 17-56-32](https://github.com/standardnotes/forum/assets/97198569/63bac134-9707-4d76-b8af-545bfce72a2e)
![Screenshot from 2023-06-30 18-12-55](https://github.com/standardnotes/forum/assets/97198569/993add59-5aa6-407b-98f9-f24e73c16b04)

See how easy it is to identify which note belongs to which indented bullet point? I need this badly. It would be the biggest UX improvement when all your notes look like this.

**Comment by user4-33** [comment]
I do not know about the initial motivation of Super Notes, but for me, it feels a bit inconvenient to always write a `/` for every text formatting action. What was the deal with those "hidden" options?

My suggestion would be a (hideable) toolbar with assigned shortcuts, similar to the richt-text editor for example.

This would allow a faster workflow than always typing slashes and scrolling down to the desired options, as well as it would improve Super Notes' accessibility for users (like me) who joined from Evernote or other less abstract environments and enjoy the advances of Super Notes compared to rich-text.

**Comment by effieeee** [comment]
[The super editor has no way of removing/hiding completed tasks](https://github.com/standardnotes/forum/discussions/2636#discussioncomment-6421012)

**Comment by josh-branco** [comment]
> * [ ]  Add option to highlight text (yellow color)
> * [ ]  Ability to resize images

The ability to highlight text would be huge!

**Reply by SpaceRaccoonFluff** [reply]
The lack of highlighting feature that is so standard elsewhere is the biggest reason why I haven't started using the SuperNotes editor yet. All my markdown notes are color code highlighted. Do the developers have a timeline for when highlighting is added?

**Comment by user4-33** [comment]
Being able to only format `some parts of a paragraph as code` – and not necessarily the whole one.

**Comment by SpaceRaccoonFluff** [comment]
- [ ] Highlight Notes In Multiple Colors (orange, red, green, blue besides just yellow)
- [ ] Highlight Text In Multiple Colors

Also thank you guys for adding the ability to highlight notes in at least one color, however its not very useful yet because when you search for a note, it highlights it in yellow so if lots of your notes are highlighted yellow, you'll never be able to find anything using the search bar. At least one other color but preferably how you can select any just like with the Rich Text editor would be nice.

**Comment by nunesgh** [comment]
- [ ] Math Mode. For instance, Joplin's [Math Mode](https://joplinapp.org/plugins/plugin/plugin.calebjohn.MathMode/) based on [mathjs](https://mathjs.org/).

**Comment by mikedolx** [comment]
Hi,

Not sure if this thread is still active. I was wondering if it would be possible to enhance the @-Linking in the super editor in a way that also new notes can be created and automatically linked to the note.

**Comment by quickllama** [comment]
Feature Request: **Shell** scripts (bash, sh, tcsh, etc.) **supported in Code Blocks**

Would be great to have syntactic highlighting for shell scripts in Super note code blocks. It looks like this has been requested a few times previously (#3044 & #3452). 

Admittedly a newbie here; according to Claude the underlying library is Highlight.js, which does have support for shell languages.

Thank you!
